libs/librepcb/project/boards/boardgerberexport.cpp

```diff
@@ -331,8 +331,33 @@ int BoardGerberExport::drawPthDrills(ExcellonGenerator& gen) const {
     foreach (const BI_FootprintPad* pad, footprint.getPads()) {
       const library::FootprintPad& libPad = pad->getLibPad();
       if (libPad.getBoardSide() == library::FootprintPad::BoardSide::THT) {
-        gen.drill(pad->getPosition(),
-                  PositiveLength(*libPad.getDrillDiameter()));  // can throw
+        if (*libPad.getDrillWidth() == *libPad.getDrillHeight()) {
+          gen.drill(pad->getPosition(), libPad.getDrillWidth());
+        }
+        else {
+          if (libPad.getWidth() > libPad.getDrillWidth()
+              and libPad.getHeight() > libPad.getDrillHeight()) {
```

Comment:
I would remove this check. If the drill is larger than the pad size, the generated Excellon file is wrong anyway, no matter if the drills are exported or not. IMHO it's a matter of the ERC to generate an error in that case, but the Excellon generator should just export the actual underlying data instead of applying some "magic" which doesn't fix the issue either.

Reply:
In such a case, where the input is broken, no output can be correct.
However, I'd rather generate sane output, which could have been correct given other input, than generate nonsensical output for which it is unclear what the board house will do

Reply:
I see your point. But when sending nonsensical output to the board house, there is a chance to get feedback from the board house that your file is nonsensical ;) When silently removing the drills, the output looks OK and the board house will probably just manufacture it, then you get wrong boards (and loose money) for almost sure.

Of course I would not generate *invalid* output, i.e. invalid Excellon files. But in case of too large drills, the file is totally fine, it just doesn't make lot of sense.